packages/variable-resolver/src/browser/variable.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { injectable, inject } from 'inversify';
+import { ILogger, Disposable, DisposableCollection } from '@theia/core';
+
+/**
+ * Variable can be used inside of strings using ${variableName} syntax.
+ */
+export interface Variable {
+
+    /**
+     * A unique name of this variable.
+     */
+    name: string;
+
+    /**
+     * A human-readable description of this variable.
+     */
+    description?: string;
+
+    /**
+     * Resolve to a string value of this variable or
+     * 'undefined' if variable cannot be resolved.
+     * Never reject.
+     */
+    resolve(): Promise<string | undefined>;
```

Comment:
You can use `MaybePromise` here, that implementors does not deal with promises. You can resolve a maybe promise with `await` syntax or `Promise.resolve`.

Reply:
Yeah, with `MaybePromise` it looks more flexible. I've updated it and also slightly refactored the `VariableResolverService`.
Thanks for the advice.